### DISCUSSION TITLE
Automatically reduce thumbnail image size on Quarto website

### DISCUSSION THREAD

**Discussion by matthew-law** [quarto-dev/quarto-cli#3793] [Q&A]
I've set up a website using the [Quarto blog template](https://quarto.org/docs/websites/website-blog.html), which includes a list of several posts and thumbnails for each of the posts. By default the thumbnails are taken from the first image included the post, if there is no `image:` specified in the YAML-like section at the start of the post's .qmd file. This works great, but many of the images I use in are very high definition (thousands of px across) – this is what I want in the posts themselves, but means that even on fast internet the website's homepage can take a while to load all the images.

Is there any automatic way to resize each of the thumbnail images so they're at most a few hundred px across?

If there isn't already a simple way to do this, I could write a script that found the thumbnail image for each post, created a resized version, saved it to the directory for the post, then specified that thumbnail as the `image:` for the post – I think I could get that to run using the [`pre-render:` project option](https://quarto.org/docs/projects/scripts.html#pre-and-post-render). But thought it worth checking there isn't already an easier way to achieve the same thing!

**Comment by danmackinlay** [comment]
#9390 is another thread on the same theme.

did you ever solve this @matthew-law ?

**Reply by matthew-law** [reply]
Thanks @danmackinlay and no, I never did solve it (or get around to writing something to downsample the images myself) – keen to hear if others develop ways to do what I'm looking for (either as part of core Quarto or as an extension)!
